Assets/SceneChangeDoor.cs

```diff
@@ -0,0 +1,59 @@
+using System;
+using UnityEngine;
+using UnityEngine.InputSystem;
+using UnityEngine.SceneManagement;
+
+public class SceneChangeDoor : MonoBehaviour
+{
+    public GameObject player;
+    public string targetScene;
+    private bool playerInInteractionArea = false;
+    private InputControls input = null;
+
+    private void Awake()
+    {
+        input = new InputControls();
+        input.Enable();
+    }
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        
+    }
+
+    private void OnTriggerEnter(Collider other)
+    {
+        Debug.Log($"Door interaction zone entered by {other.name}");
+        if (other.gameObject == player)
```

Comment:
I suggest checking tag or existence of the player script and not an object reference. I think that would be more reliable / stable.

Reply:
We currently use tags for this, e.g. https://github.com/sllewely/capitalismgame/blob/7a1abbea7633f9b67622e8a3b93546ad11b5ebb5/Assets/Scripts/Inventory/PickupableBehavior.cs#L22

Tags are basically a hardcoded enum set up in the Unity editor.